src/SOS/Strike/gcroot.cpp

```diff
@@ -1099,7 +1099,7 @@ size_t GCRootImpl::GetSizeOfObject(TADDR obj, MTInfo *info)
     {
         // this is an array, so the size has to include the size of the components. We read the number
         // of components from the target and multiply by the component size to get the size.
-        DWORD components = GetComponents(obj, info->MethodTable);
+        size_t components = GetComponents(obj, info->MethodTable);
         res += info->ComponentSize * components;
```

Comment:
The multiplications like this one were the source of the bug. Both ComponentSize and components can be <4GB, but the result can be more than 4GB.

Reply:
Does dumpheap have any similar problems? I haven't looked myself.